src/test/java/com/connexta/store/adaptors/S3StorageAdaptorITests.java

```diff
@@ -153,11 +157,42 @@ public void testStoreWithContentLengthNotMatching() {
               10L,
               MediaType.APPLICATION_XML_VALUE,
               new ByteArrayInputStream(ASDF.getBytes()),
-              KEY,
+              DATASET_ID,
               Map.of());
         });
   }
 
+  @Test
+  void testUpdatingStoreStatus() {
+    // Store a file and set status as staged
+    storageAdaptor.store(
+        4L,
+        MediaType.APPLICATION_XML_VALUE,
+        new ByteArrayInputStream(ASDF.getBytes()),
+        DATASET_ID,
+        Map.of());
+    assertEquals(STAGED, getStatusTag(DATASET_ID));
+
+    // Promote file to stored status
+    storageAdaptor.updateStatus(DATASET_ID, STORED);
+    assertEquals(STORED, getStatusTag(DATASET_ID));
```

Comment:
✏️ Should we test that the file is still retrievable after changing the status?

Reply:
I think that kind of test is better suited for the StoreService where the logic to control what can and cannot be retrieved will live.

Reply:
Updating status should not change the file that was stored. We don't currently have any tests for that.